// css-property-names/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrefixedCSSPropertyNames = exports.CSSPropertyNames = void 0;
exports.CSSPropertyNames = [
    'accentColor',
    'additiveSymbols',
    'alignContent',
    'alignItems',
    'alignSelf',
    'alignmentBaseline',
    'all',
    'animation',
    'animationComposition',
    'animationDelay',
    'animationDirection',
    'animationDuration',
    'animationFillMode',
    'animationIterationCount',
    'animationName',
    'animationPlayState',
    'animationTimingFunction',
    'appRegion',
    'appearance',
    'ascentOverride',
    'aspectRatio',
    'backdropFilter',
    'backfaceVisibility',
    'background',
    'backgroundAttachment',
    'backgroundBlendMode',
    'backgroundClip',
    'backgroundColor',
    'backgroundImage',
    'backgroundOrigin',
    'backgroundPosition',
    'backgroundPositionX',
    'backgroundPositionY',
    'backgroundRepeat',
    'backgroundRepeatX',
    'backgroundRepeatY',
    'backgroundSize',
    'basePalette',
    'baselineShift',
    'baselineSource',
    'blockSize',
    'border',
    'borderBlock',
    'borderBlockColor',
    'borderBlockEnd',
    'borderBlockEndColor',
    'borderBlockEndStyle',
    'borderBlockEndWidth',
    'borderBlockStart',
    'borderBlockStartColor',
    'borderBlockStartStyle',
    'borderBlockStartWidth',
    'borderBlockStyle',
    'borderBlockWidth',
    'borderBottom',
    'borderBottomColor',
    'borderBottomLeftRadius',
    'borderBottomRightRadius',
    'borderBottomStyle',
    'borderBottomWidth',
    'borderCollapse',
    'borderColor',
    'borderEndEndRadius',
    'borderEndStartRadius',
    'borderImage',
    'borderImageOutset',
    'borderImageRepeat',
    'borderImageSlice',
    'borderImageSource',
    'borderImageWidth',
    'borderInline',
    'borderInlineColor',
    'borderInlineEnd',
    'borderInlineEndColor',
    'borderInlineEndStyle',
    'borderInlineEndWidth',
    'borderInlineStart',
    'borderInlineStartColor',
    'borderInlineStartStyle',
    'borderInlineStartWidth',
    'borderInlineStyle',
    'borderInlineWidth',
    'borderLeft',
    'borderLeftColor',
    'borderLeftStyle',
    'borderLeftWidth',
    'borderRadius',
    'borderRight',
    'borderRightColor',
    'borderRightStyle',
    'borderRightWidth',
    'borderSpacing',
    'borderStartEndRadius',
    'borderStartStartRadius',
    'borderStyle',
    'borderTop',
    'borderTopColor',
    'borderTopLeftRadius',
    'borderTopRightRadius',
    'borderTopStyle',
    'borderTopWidth',
    'borderWidth',
    'bottom',
    'boxShadow',
    'boxSizing',
    'breakAfter',
    'breakBefore',
    'breakInside',
    'bufferedRendering',
    'captionSide',
    'caretColor',
    'clear',
    'clip',
    'clipPath',
    'clipRule',
    'color',
    'colorInterpolation',
    'colorInterpolationFilters',
    'colorRendering',
    'colorScheme',
    'columnCount',
    'columnFill',
    'columnGap',
    'columnRule',
    'columnRuleColor',
    'columnRuleStyle',
    'columnRuleWidth',
    'columnSpan',
    'columnWidth',
    'columns',
    'contain',
    'containIntrinsicBlockSize',
    'containIntrinsicHeight',
    'containIntrinsicInlineSize',
    'containIntrinsicSize',
    'containIntrinsicWidth',
    'container',
    'containerName',
    'containerType',
    'content',
    'contentVisibility',
    'counterIncrement',
    'counterReset',
    'counterSet',
    'cursor',
    'cx',
    'cy',
    'd',
    'descentOverride',
    'direction',
    'display',
    'dominantBaseline',
    'emptyCells',
    'fallback',
    'fill',
    'fillOpacity',
    'fillRule',
    'filter',
    'flex',
    'flexBasis',
    'flexDirection',
    'flexFlow',
    'flexGrow',
    'flexShrink',
    'flexWrap',
    'float',
    'floodColor',
    'floodOpacity',
    'font',
    'fontDisplay',
    'fontFamily',
    'fontFeatureSettings',
    'fontKerning',
    'fontOpticalSizing',
    'fontPalette',
    'fontSize',
    'fontStretch',
    'fontStyle',
    'fontSynthesis',
    'fontSynthesisSmallCaps',
    'fontSynthesisStyle',
    'fontSynthesisWeight',
    'fontVariant',
    'fontVariantAlternates',
    'fontVariantCaps',
    'fontVariantEastAsian',
    'fontVariantLigatures',
    'fontVariantNumeric',
    'fontVariationSettings',
    'fontWeight',
    'forcedColorAdjust',
    'gap',
    'grid',
    'gridArea',
    'gridAutoColumns',
    'gridAutoFlow',
    'gridAutoRows',
    'gridColumn',
    'gridColumnEnd',
    'gridColumnGap',
    'gridColumnStart',
    'gridGap',
    'gridRow',
    'gridRowEnd',
    'gridRowGap',
    'gridRowStart',
    'gridTemplate',
    'gridTemplateAreas',
    'gridTemplateColumns',
    'gridTemplateRows',
    'height',
    'hyphenateCharacter',
    'hyphenateLimitChars',
    'hyphens',
    'imageOrientation',
    'imageRendering',
    'inherits',
    'initialLetter',
    'initialValue',
    'inlineSize',
    'inset',
    'insetBlock',
    'insetBlockEnd',
    'insetBlockStart',
    'insetInline',
    'insetInlineEnd',
    'insetInlineStart',
    'isolation',
    'justifyContent',
    'justifyItems',
    'justifySelf',
    'left',
    'letterSpacing',
    'lightingColor',
    'lineBreak',
    'lineGapOverride',
    'lineHeight',
    'listStyle',
    'listStyleImage',
    'listStylePosition',
    'listStyleType',
    'margin',
    'marginBlock',
    'marginBlockEnd',
    'marginBlockStart',
    'marginBottom',
    'marginInline',
    'marginInlineEnd',
    'marginInlineStart',
    'marginLeft',
    'marginRight',
    'marginTop',
    'marker',
    'markerEnd',
    'markerMid',
    'markerStart',
    'mask',
    'maskType',
    'mathDepth',
    'mathShift',
    'mathStyle',
    'maxBlockSize',
    'maxHeight',
    'maxInlineSize',
    'maxWidth',
    'minBlockSize',
    'minHeight',
    'minInlineSize',
    'minWidth',
    'mixBlendMode',
    'negative',
    'objectFit',
    'objectPosition',
    'objectViewBox',
    'offset',
    'offsetDistance',
    'offsetPath',
    'offsetRotate',
    'opacity',
    'order',
    'orphans',
    'outline',
    'outlineColor',
    'outlineOffset',
    'outlineStyle',
    'outlineWidth',
    'overflow',
    'overflowAnchor',
    'overflowClipMargin',
    'overflowWrap',
    'overflowX',
    'overflowY',
    'overrideColors',
    'overscrollBehavior',
    'overscrollBehaviorBlock',
    'overscrollBehaviorInline',
    'overscrollBehaviorX',
    'overscrollBehaviorY',
    'pad',
    'padding',
    'paddingBlock',
    'paddingBlockEnd',
    'paddingBlockStart',
    'paddingBottom',
    'paddingInline',
    'paddingInlineEnd',
    'paddingInlineStart',
    'paddingLeft',
    'paddingRight',
    'paddingTop',
    'page',
    'pageBreakAfter',
    'pageBreakBefore',
    'pageBreakInside',
    'pageOrientation',
    'paintOrder',
    'perspective',
    'perspectiveOrigin',
    'placeContent',
    'placeItems',
    'placeSelf',
    'pointerEvents',
    'position',
    'prefix',
    'quotes',
    'r',
    'range',
    'resize',
    'right',
    'rotate',
    'rowGap',
    'rubyPosition',
    'rx',
    'ry',
    'scale',
    'scrollBehavior',
    'scrollMargin',
    'scrollMarginBlock',
    'scrollMarginBlockEnd',
    'scrollMarginBlockStart',
    'scrollMarginBottom',
    'scrollMarginInline',
    'scrollMarginInlineEnd',
    'scrollMarginInlineStart',
    'scrollMarginLeft',
    'scrollMarginRight',
    'scrollMarginTop',
    'scrollPadding',
    'scrollPaddingBlock',
    'scrollPaddingBlockEnd',
    'scrollPaddingBlockStart',
    'scrollPaddingBottom',
    'scrollPaddingInline',
    'scrollPaddingInlineEnd',
    'scrollPaddingInlineStart',
    'scrollPaddingLeft',
    'scrollPaddingRight',
    'scrollPaddingTop',
    'scrollSnapAlign',
    'scrollSnapStop',
    'scrollSnapType',
    'scrollbarGutter',
    'shapeImageThreshold',
    'shapeMargin',
    'shapeOutside',
    'shapeRendering',
    'size',
    'sizeAdjust',
    'speak',
    'speakAs',
    'src',
    'stopColor',
    'stopOpacity',
    'stroke',
    'strokeDasharray',
    'strokeDashoffset',
    'strokeLinecap',
    'strokeLinejoin',
    'strokeMiterlimit',
    'strokeOpacity',
    'strokeWidth',
    'suffix',
    'symbols',
    'syntax',
    'system',
    'tabSize',
    'tableLayout',
    'textAlign',
    'textAlignLast',
    'textAnchor',
    'textCombineUpright',
    'textDecoration',
    'textDecorationColor',
    'textDecorationLine',
    'textDecorationSkipInk',
    'textDecorationStyle',
    'textDecorationThickness',
    'textEmphasis',
    'textEmphasisColor',
    'textEmphasisPosition',
    'textEmphasisStyle',
    'textIndent',
    'textOrientation',
    'textOverflow',
    'textRendering',
    'textShadow',
    'textSizeAdjust',
    'textTransform',
    'textUnderlineOffset',
    'textUnderlinePosition',
    'top',
    'touchAction',
    'transform',
    'transformBox',
    'transformOrigin',
    'transformStyle',
    'transition',
    'transitionDelay',
    'transitionDuration',
    'transitionProperty',
    'transitionTimingFunction',
    'translate',
    'unicodeBidi',
    'unicodeRange',
    'userSelect',
    'vectorEffect',
    'verticalAlign',
    'viewTransitionName',
    'visibility',
    'webkitAlignContent',
    'webkitAlignItems',
    'webkitAlignSelf',
    'webkitAnimation',
    'webkitAnimationDelay',
    'webkitAnimationDirection',
    'webkitAnimationDuration',
    'webkitAnimationFillMode',
    'webkitAnimationIterationCount',
    'webkitAnimationName',
    'webkitAnimationPlayState',
    'webkitAnimationTimingFunction',
    'webkitAppRegion',
    'webkitAppearance',
    'webkitBackfaceVisibility',
    'webkitBackgroundClip',
    'webkitBackgroundOrigin',
    'webkitBackgroundSize',
    'webkitBorderAfter',
    'webkitBorderAfterColor',
    'webkitBorderAfterStyle',
    'webkitBorderAfterWidth',
    'webkitBorderBefore',
    'webkitBorderBeforeColor',
    'webkitBorderBeforeStyle',
    'webkitBorderBeforeWidth',
    'webkitBorderBottomLeftRadius',
    'webkitBorderBottomRightRadius',
    'webkitBorderEnd',
    'webkitBorderEndColor',
    'webkitBorderEndStyle',
    'webkitBorderEndWidth',
    'webkitBorderHorizontalSpacing',
    'webkitBorderImage',
    'webkitBorderRadius',
    'webkitBorderStart',
    'webkitBorderStartColor',
    'webkitBorderStartStyle',
    'webkitBorderStartWidth',
    'webkitBorderTopLeftRadius',
    'webkitBorderTopRightRadius',
    'webkitBorderVerticalSpacing',
    'webkitBoxAlign',
    'webkitBoxDecorationBreak',
    'webkitBoxDirection',
    'webkitBoxFlex',
    'webkitBoxOrdinalGroup',
    'webkitBoxOrient',
    'webkitBoxPack',
    'webkitBoxReflect',
    'webkitBoxShadow',
    'webkitBoxSizing',
    'webkitClipPath',
    'webkitColumnBreakAfter',
    'webkitColumnBreakBefore',
    'webkitColumnBreakInside',
    'webkitColumnCount',
    'webkitColumnGap',
    'webkitColumnRule',
    'webkitColumnRuleColor',
    'webkitColumnRuleStyle',
    'webkitColumnRuleWidth',
    'webkitColumnSpan',
    'webkitColumnWidth',
    'webkitColumns',
    'webkitFilter',
    'webkitFlex',
    'webkitFlexBasis',
    'webkitFlexDirection',
    'webkitFlexFlow',
    'webkitFlexGrow',
    'webkitFlexShrink',
    'webkitFlexWrap',
    'webkitFontFeatureSettings',
    'webkitFontSmoothing',
    'webkitHighlight',
    'webkitHyphenateCharacter',
    'webkitJustifyContent',
    'webkitLineBreak',
    'webkitLineClamp',
    'webkitLocale',
    'webkitLogicalHeight',
    'webkitLogicalWidth',
    'webkitMarginAfter',
    'webkitMarginBefore',
    'webkitMarginEnd',
    'webkitMarginStart',
    'webkitMask',
    'webkitMaskBoxImage',
    'webkitMaskBoxImageOutset',
    'webkitMaskBoxImageRepeat',
    'webkitMaskBoxImageSlice',
    'webkitMaskBoxImageSource',
    'webkitMaskBoxImageWidth',
    'webkitMaskClip',
    'webkitMaskComposite',
    'webkitMaskImage',
    'webkitMaskOrigin',
    'webkitMaskPosition',
    'webkitMaskPositionX',
    'webkitMaskPositionY',
    'webkitMaskRepeat',
    'webkitMaskRepeatX',
    'webkitMaskRepeatY',
    'webkitMaskSize',
    'webkitMaxLogicalHeight',
    'webkitMaxLogicalWidth',
    'webkitMinLogicalHeight',
    'webkitMinLogicalWidth',
    'webkitOpacity',
    'webkitOrder',
    'webkitPaddingAfter',
    'webkitPaddingBefore',
    'webkitPaddingEnd',
    'webkitPaddingStart',
    'webkitPerspective',
    'webkitPerspectiveOrigin',
    'webkitPerspectiveOriginX',
    'webkitPerspectiveOriginY',
    'webkitPrintColorAdjust',
    'webkitRtlOrdering',
    'webkitRubyPosition',
    'webkitShapeImageThreshold',
    'webkitShapeMargin',
    'webkitShapeOutside',
    'webkitTapHighlightColor',
    'webkitTextCombine',
    'webkitTextDecorationsInEffect',
    'webkitTextEmphasis',
    'webkitTextEmphasisColor',
    'webkitTextEmphasisPosition',
    'webkitTextEmphasisStyle',
    'webkitTextFillColor',
    'webkitTextOrientation',
    'webkitTextSecurity',
    'webkitTextSizeAdjust',
    'webkitTextStroke',
    'webkitTextStrokeColor',
    'webkitTextStrokeWidth',
    'webkitTransform',
    'webkitTransformOrigin',
    'webkitTransformOriginX',
    'webkitTransformOriginY',
    'webkitTransformOriginZ',
    'webkitTransformStyle',
    'webkitTransition',
    'webkitTransitionDelay',
    'webkitTransitionDuration',
    'webkitTransitionProperty',
    'webkitTransitionTimingFunction',
    'webkitUserDrag',
    'webkitUserModify',
    'webkitUserSelect',
    'webkitWritingMode',
    'whiteSpace',
    'widows',
    'width',
    'willChange',
    'wordBreak',
    'wordSpacing',
    'wordWrap',
    'writingMode',
    'x',
    'y',
    'zIndex',
    'zoom'
];
exports.PrefixedCSSPropertyNames = {
    cssAccentColor: 'accentColor',
    cssAdditiveSymbols: 'additiveSymbols',
    cssAlignContent: 'alignContent',
    cssAlignItems: 'alignItems',
    cssAlignSelf: 'alignSelf',
    cssAlignmentBaseline: 'alignmentBaseline',
    cssAll: 'all',
    cssAnimation: 'animation',
    cssAnimationComposition: 'animationComposition',
    cssAnimationDelay: 'animationDelay',
    cssAnimationDirection: 'animationDirection',
    cssAnimationDuration: 'animationDuration',
    cssAnimationFillMode: 'animationFillMode',
    cssAnimationIterationCount: 'animationIterationCount',
    cssAnimationName: 'animationName',
    cssAnimationPlayState: 'animationPlayState',
    cssAnimationTimingFunction: 'animationTimingFunction',
    cssAppRegion: 'appRegion',
    cssAppearance: 'appearance',
    cssAscentOverride: 'ascentOverride',
    cssAspectRatio: 'aspectRatio',
    cssBackdropFilter: 'backdropFilter',
    cssBackfaceVisibility: 'backfaceVisibility',
    cssBackground: 'background',
    cssBackgroundAttachment: 'backgroundAttachment',
    cssBackgroundBlendMode: 'backgroundBlendMode',
    cssBackgroundClip: 'backgroundClip',
    cssBackgroundColor: 'backgroundColor',
    cssBackgroundImage: 'backgroundImage',
    cssBackgroundOrigin: 'backgroundOrigin',
    cssBackgroundPosition: 'backgroundPosition',
    cssBackgroundPositionX: 'backgroundPositionX',
    cssBackgroundPositionY: 'backgroundPositionY',
    cssBackgroundRepeat: 'backgroundRepeat',
    cssBackgroundRepeatX: 'backgroundRepeatX',
    cssBackgroundRepeatY: 'backgroundRepeatY',
    cssBackgroundSize: 'backgroundSize',
    cssBasePalette: 'basePalette',
    cssBaselineShift: 'baselineShift',
    cssBaselineSource: 'baselineSource',
    cssBlockSize: 'blockSize',
    cssBorder: 'border',
    cssBorderBlock: 'borderBlock',
    cssBorderBlockColor: 'borderBlockColor',
    cssBorderBlockEnd: 'borderBlockEnd',
    cssBorderBlockEndColor: 'borderBlockEndColor',
    cssBorderBlockEndStyle: 'borderBlockEndStyle',
    cssBorderBlockEndWidth: 'borderBlockEndWidth',
    cssBorderBlockStart: 'borderBlockStart',
    cssBorderBlockStartColor: 'borderBlockStartColor',
    cssBorderBlockStartStyle: 'borderBlockStartStyle',
    cssBorderBlockStartWidth: 'borderBlockStartWidth',
    cssBorderBlockStyle: 'borderBlockStyle',
    cssBorderBlockWidth: 'borderBlockWidth',
    cssBorderBottom: 'borderBottom',
    cssBorderBottomColor: 'borderBottomColor',
    cssBorderBottomLeftRadius: 'borderBottomLeftRadius',
    cssBorderBottomRightRadius: 'borderBottomRightRadius',
    cssBorderBottomStyle: 'borderBottomStyle',
    cssBorderBottomWidth: 'borderBottomWidth',
    cssBorderCollapse: 'borderCollapse',
    cssBorderColor: 'borderColor',
    cssBorderEndEndRadius: 'borderEndEndRadius',
    cssBorderEndStartRadius: 'borderEndStartRadius',
    cssBorderImage: 'borderImage',
    cssBorderImageOutset: 'borderImageOutset',
    cssBorderImageRepeat: 'borderImageRepeat',
    cssBorderImageSlice: 'borderImageSlice',
    cssBorderImageSource: 'borderImageSource',
    cssBorderImageWidth: 'borderImageWidth',
    cssBorderInline: 'borderInline',
    cssBorderInlineColor: 'borderInlineColor',
    cssBorderInlineEnd: 'borderInlineEnd',
    cssBorderInlineEndColor: 'borderInlineEndColor',
    cssBorderInlineEndStyle: 'borderInlineEndStyle',
    cssBorderInlineEndWidth: 'borderInlineEndWidth',
    cssBorderInlineStart: 'borderInlineStart',
    cssBorderInlineStartColor: 'borderInlineStartColor',
    cssBorderInlineStartStyle: 'borderInlineStartStyle',
    cssBorderInlineStartWidth: 'borderInlineStartWidth',
    cssBorderInlineStyle: 'borderInlineStyle',
    cssBorderInlineWidth: 'borderInlineWidth',
    cssBorderLeft: 'borderLeft',
    cssBorderLeftColor: 'borderLeftColor',
    cssBorderLeftStyle: 'borderLeftStyle',
    cssBorderLeftWidth: 'borderLeftWidth',
    cssBorderRadius: 'borderRadius',
    cssBorderRight: 'borderRight',
    cssBorderRightColor: 'borderRightColor',
    cssBorderRightStyle: 'borderRightStyle',
    cssBorderRightWidth: 'borderRightWidth',
    cssBorderSpacing: 'borderSpacing',
    cssBorderStartEndRadius: 'borderStartEndRadius',
    cssBorderStartStartRadius: 'borderStartStartRadius',
    cssBorderStyle: 'borderStyle',
    cssBorderTop: 'borderTop',
    cssBorderTopColor: 'borderTopColor',
    cssBorderTopLeftRadius: 'borderTopLeftRadius',
    cssBorderTopRightRadius: 'borderTopRightRadius',
    cssBorderTopStyle: 'borderTopStyle',
    cssBorderTopWidth: 'borderTopWidth',
    cssBorderWidth: 'borderWidth',
    cssBottom: 'bottom',
    cssBoxShadow: 'boxShadow',
    cssBoxSizing: 'boxSizing',
    cssBreakAfter: 'breakAfter',
    cssBreakBefore: 'breakBefore',
    cssBreakInside: 'breakInside',
    cssBufferedRendering: 'bufferedRendering',
    cssCaptionSide: 'captionSide',
    cssCaretColor: 'caretColor',
    cssClear: 'clear',
    cssClip: 'clip',
    cssClipPath: 'clipPath',
    cssClipRule: 'clipRule',
    cssColor: 'color',
    cssColorInterpolation: 'colorInterpolation',
    cssColorInterpolationFilters: 'colorInterpolationFilters',
    cssColorRendering: 'colorRendering',
    cssColorScheme: 'colorScheme',
    cssColumnCount: 'columnCount',
    cssColumnFill: 'columnFill',
    cssColumnGap: 'columnGap',
    cssColumnRule: 'columnRule',
    cssColumnRuleColor: 'columnRuleColor',
    cssColumnRuleStyle: 'columnRuleStyle',
    cssColumnRuleWidth: 'columnRuleWidth',
    cssColumnSpan: 'columnSpan',
    cssColumnWidth: 'columnWidth',
    cssColumns: 'columns',
    cssContain: 'contain',
    cssContainIntrinsicBlockSize: 'containIntrinsicBlockSize',
    cssContainIntrinsicHeight: 'containIntrinsicHeight',
    cssContainIntrinsicInlineSize: 'containIntrinsicInlineSize',
    cssContainIntrinsicSize: 'containIntrinsicSize',
    cssContainIntrinsicWidth: 'containIntrinsicWidth',
    cssContainer: 'container',
    cssContainerName: 'containerName',
    cssContainerType: 'containerType',
    cssContent: 'content',
    cssContentVisibility: 'contentVisibility',
    cssCounterIncrement: 'counterIncrement',
    cssCounterReset: 'counterReset',
    cssCounterSet: 'counterSet',
    cssCursor: 'cursor',
    cssCx: 'cx',
    cssCy: 'cy',
    cssD: 'd',
    cssDescentOverride: 'descentOverride',
    cssDirection: 'direction',
    cssDisplay: 'display',
    cssDominantBaseline: 'dominantBaseline',
    cssEmptyCells: 'emptyCells',
    cssFallback: 'fallback',
    cssFill: 'fill',
    cssFillOpacity: 'fillOpacity',
    cssFillRule: 'fillRule',
    cssFilter: 'filter',
    cssFlex: 'flex',
    cssFlexBasis: 'flexBasis',
    cssFlexDirection: 'flexDirection',
    cssFlexFlow: 'flexFlow',
    cssFlexGrow: 'flexGrow',
    cssFlexShrink: 'flexShrink',
    cssFlexWrap: 'flexWrap',
    cssFloat: 'float',
    cssFloodColor: 'floodColor',
    cssFloodOpacity: 'floodOpacity',
    cssFont: 'font',
    cssFontDisplay: 'fontDisplay',
    cssFontFamily: 'fontFamily',
    cssFontFeatureSettings: 'fontFeatureSettings',
    cssFontKerning: 'fontKerning',
    cssFontOpticalSizing: 'fontOpticalSizing',
    cssFontPalette: 'fontPalette',
    cssFontSize: 'fontSize',
    cssFontStretch: 'fontStretch',
    cssFontStyle: 'fontStyle',
    cssFontSynthesis: 'fontSynthesis',
    cssFontSynthesisSmallCaps: 'fontSynthesisSmallCaps',
    cssFontSynthesisStyle: 'fontSynthesisStyle',
    cssFontSynthesisWeight: 'fontSynthesisWeight',
    cssFontVariant: 'fontVariant',
    cssFontVariantAlternates: 'fontVariantAlternates',
    cssFontVariantCaps: 'fontVariantCaps',
    cssFontVariantEastAsian: 'fontVariantEastAsian',
    cssFontVariantLigatures: 'fontVariantLigatures',
    cssFontVariantNumeric: 'fontVariantNumeric',
    cssFontVariationSettings: 'fontVariationSettings',
    cssFontWeight: 'fontWeight',
    cssForcedColorAdjust: 'forcedColorAdjust',
    cssGap: 'gap',
    cssGrid: 'grid',
    cssGridArea: 'gridArea',
    cssGridAutoColumns: 'gridAutoColumns',
    cssGridAutoFlow: 'gridAutoFlow',
    cssGridAutoRows: 'gridAutoRows',
    cssGridColumn: 'gridColumn',
    cssGridColumnEnd: 'gridColumnEnd',
    cssGridColumnGap: 'gridColumnGap',
    cssGridColumnStart: 'gridColumnStart',
    cssGridGap: 'gridGap',
    cssGridRow: 'gridRow',
    cssGridRowEnd: 'gridRowEnd',
    cssGridRowGap: 'gridRowGap',
    cssGridRowStart: 'gridRowStart',
    cssGridTemplate: 'gridTemplate',
    cssGridTemplateAreas: 'gridTemplateAreas',
    cssGridTemplateColumns: 'gridTemplateColumns',
    cssGridTemplateRows: 'gridTemplateRows',
    cssHeight: 'height',
    cssHyphenateCharacter: 'hyphenateCharacter',
    cssHyphenateLimitChars: 'hyphenateLimitChars',
    cssHyphens: 'hyphens',
    cssImageOrientation: 'imageOrientation',
    cssImageRendering: 'imageRendering',
    cssInherits: 'inherits',
    cssInitialLetter: 'initialLetter',
    cssInitialValue: 'initialValue',
    cssInlineSize: 'inlineSize',
    cssInset: 'inset',
    cssInsetBlock: 'insetBlock',
    cssInsetBlockEnd: 'insetBlockEnd',
    cssInsetBlockStart: 'insetBlockStart',
    cssInsetInline: 'insetInline',
    cssInsetInlineEnd: 'insetInlineEnd',
    cssInsetInlineStart: 'insetInlineStart',
    cssIsolation: 'isolation',
    cssJustifyContent: 'justifyContent',
    cssJustifyItems: 'justifyItems',
    cssJustifySelf: 'justifySelf',
    cssLeft: 'left',
    cssLetterSpacing: 'letterSpacing',
    cssLightingColor: 'lightingColor',
    cssLineBreak: 'lineBreak',
    cssLineGapOverride: 'lineGapOverride',
    cssLineHeight: 'lineHeight',
    cssListStyle: 'listStyle',
    cssListStyleImage: 'listStyleImage',
    cssListStylePosition: 'listStylePosition',
    cssListStyleType: 'listStyleType',
    cssMargin: 'margin',
    cssMarginBlock: 'marginBlock',
    cssMarginBlockEnd: 'marginBlockEnd',
    cssMarginBlockStart: 'marginBlockStart',
    cssMarginBottom: 'marginBottom',
    cssMarginInline: 'marginInline',
    cssMarginInlineEnd: 'marginInlineEnd',
    cssMarginInlineStart: 'marginInlineStart',
    cssMarginLeft: 'marginLeft',
    cssMarginRight: 'marginRight',
    cssMarginTop: 'marginTop',
    cssMarker: 'marker',
    cssMarkerEnd: 'markerEnd',
    cssMarkerMid: 'markerMid',
    cssMarkerStart: 'markerStart',
    cssMask: 'mask',
    cssMaskType: 'maskType',
    cssMathDepth: 'mathDepth',
    cssMathShift: 'mathShift',
    cssMathStyle: 'mathStyle',
    cssMaxBlockSize: 'maxBlockSize',
    cssMaxHeight: 'maxHeight',
    cssMaxInlineSize: 'maxInlineSize',
    cssMaxWidth: 'maxWidth',
    cssMinBlockSize: 'minBlockSize',
    cssMinHeight: 'minHeight',
    cssMinInlineSize: 'minInlineSize',
    cssMinWidth: 'minWidth',
    cssMixBlendMode: 'mixBlendMode',
    cssNegative: 'negative',
    cssObjectFit: 'objectFit',
    cssObjectPosition: 'objectPosition',
    cssObjectViewBox: 'objectViewBox',
    cssOffset: 'offset',
    cssOffsetDistance: 'offsetDistance',
    cssOffsetPath: 'offsetPath',
    cssOffsetRotate: 'offsetRotate',
    cssOpacity: 'opacity',
    cssOrder: 'order',
    cssOrphans: 'orphans',
    cssOutline: 'outline',
    cssOutlineColor: 'outlineColor',
    cssOutlineOffset: 'outlineOffset',
    cssOutlineStyle: 'outlineStyle',
    cssOutlineWidth: 'outlineWidth',
    cssOverflow: 'overflow',
    cssOverflowAnchor: 'overflowAnchor',
    cssOverflowClipMargin: 'overflowClipMargin',
    cssOverflowWrap: 'overflowWrap',
    cssOverflowX: 'overflowX',
    cssOverflowY: 'overflowY',
    cssOverrideColors: 'overrideColors',
    cssOverscrollBehavior: 'overscrollBehavior',
    cssOverscrollBehaviorBlock: 'overscrollBehaviorBlock',
    cssOverscrollBehaviorInline: 'overscrollBehaviorInline',
    cssOverscrollBehaviorX: 'overscrollBehaviorX',
    cssOverscrollBehaviorY: 'overscrollBehaviorY',
    cssPad: 'pad',
    cssPadding: 'padding',
    cssPaddingBlock: 'paddingBlock',
    cssPaddingBlockEnd: 'paddingBlockEnd',
    cssPaddingBlockStart: 'paddingBlockStart',
    cssPaddingBottom: 'paddingBottom',
    cssPaddingInline: 'paddingInline',
    cssPaddingInlineEnd: 'paddingInlineEnd',
    cssPaddingInlineStart: 'paddingInlineStart',
    cssPaddingLeft: 'paddingLeft',
    cssPaddingRight: 'paddingRight',
    cssPaddingTop: 'paddingTop',
    cssPage: 'page',
    cssPageBreakAfter: 'pageBreakAfter',
    cssPageBreakBefore: 'pageBreakBefore',
    cssPageBreakInside: 'pageBreakInside',
    cssPageOrientation: 'pageOrientation',
    cssPaintOrder: 'paintOrder',
    cssPerspective: 'perspective',
    cssPerspectiveOrigin: 'perspectiveOrigin',
    cssPlaceContent: 'placeContent',
    cssPlaceItems: 'placeItems',
    cssPlaceSelf: 'placeSelf',
    cssPointerEvents: 'pointerEvents',
    cssPosition: 'position',
    cssPrefix: 'prefix',
    cssQuotes: 'quotes',
    cssR: 'r',
    cssRange: 'range',
    cssResize: 'resize',
    cssRight: 'right',
    cssRotate: 'rotate',
    cssRowGap: 'rowGap',
    cssRubyPosition: 'rubyPosition',
    cssRx: 'rx',
    cssRy: 'ry',
    cssScale: 'scale',
    cssScrollBehavior: 'scrollBehavior',
    cssScrollMargin: 'scrollMargin',
    cssScrollMarginBlock: 'scrollMarginBlock',
    cssScrollMarginBlockEnd: 'scrollMarginBlockEnd',
    cssScrollMarginBlockStart: 'scrollMarginBlockStart',
    cssScrollMarginBottom: 'scrollMarginBottom',
    cssScrollMarginInline: 'scrollMarginInline',
    cssScrollMarginInlineEnd: 'scrollMarginInlineEnd',
    cssScrollMarginInlineStart: 'scrollMarginInlineStart',
    cssScrollMarginLeft: 'scrollMarginLeft',
    cssScrollMarginRight: 'scrollMarginRight',
    cssScrollMarginTop: 'scrollMarginTop',
    cssScrollPadding: 'scrollPadding',
    cssScrollPaddingBlock: 'scrollPaddingBlock',
    cssScrollPaddingBlockEnd: 'scrollPaddingBlockEnd',
    cssScrollPaddingBlockStart: 'scrollPaddingBlockStart',
    cssScrollPaddingBottom: 'scrollPaddingBottom',
    cssScrollPaddingInline: 'scrollPaddingInline',
    cssScrollPaddingInlineEnd: 'scrollPaddingInlineEnd',
    cssScrollPaddingInlineStart: 'scrollPaddingInlineStart',
    cssScrollPaddingLeft: 'scrollPaddingLeft',
    cssScrollPaddingRight: 'scrollPaddingRight',
    cssScrollPaddingTop: 'scrollPaddingTop',
    cssScrollSnapAlign: 'scrollSnapAlign',
    cssScrollSnapStop: 'scrollSnapStop',
    cssScrollSnapType: 'scrollSnapType',
    cssScrollbarGutter: 'scrollbarGutter',
    cssShapeImageThreshold: 'shapeImageThreshold',
    cssShapeMargin: 'shapeMargin',
    cssShapeOutside: 'shapeOutside',
    cssShapeRendering: 'shapeRendering',
    cssSize: 'size',
    cssSizeAdjust: 'sizeAdjust',
    cssSpeak: 'speak',
    cssSpeakAs: 'speakAs',
    cssSrc: 'src',
    cssStopColor: 'stopColor',
    cssStopOpacity: 'stopOpacity',
    cssStroke: 'stroke',
    cssStrokeDasharray: 'strokeDasharray',
    cssStrokeDashoffset: 'strokeDashoffset',
    cssStrokeLinecap: 'strokeLinecap',
    cssStrokeLinejoin: 'strokeLinejoin',
    cssStrokeMiterlimit: 'strokeMiterlimit',
    cssStrokeOpacity: 'strokeOpacity',
    cssStrokeWidth: 'strokeWidth',
    cssSuffix: 'suffix',
    cssSymbols: 'symbols',
    cssSyntax: 'syntax',
    cssSystem: 'system',
    cssTabSize: 'tabSize',
    cssTableLayout: 'tableLayout',
    cssTextAlign: 'textAlign',
    cssTextAlignLast: 'textAlignLast',
    cssTextAnchor: 'textAnchor',
    cssTextCombineUpright: 'textCombineUpright',
    cssTextDecoration: 'textDecoration',
    cssTextDecorationColor: 'textDecorationColor',
    cssTextDecorationLine: 'textDecorationLine',
    cssTextDecorationSkipInk: 'textDecorationSkipInk',
    cssTextDecorationStyle: 'textDecorationStyle',
    cssTextDecorationThickness: 'textDecorationThickness',
    cssTextEmphasis: 'textEmphasis',
    cssTextEmphasisColor: 'textEmphasisColor',
    cssTextEmphasisPosition: 'textEmphasisPosition',
    cssTextEmphasisStyle: 'textEmphasisStyle',
    cssTextIndent: 'textIndent',
    cssTextOrientation: 'textOrientation',
    cssTextOverflow: 'textOverflow',
    cssTextRendering: 'textRendering',
    cssTextShadow: 'textShadow',
    cssTextSizeAdjust: 'textSizeAdjust',
    cssTextTransform: 'textTransform',
    cssTextUnderlineOffset: 'textUnderlineOffset',
    cssTextUnderlinePosition: 'textUnderlinePosition',
    cssTop: 'top',
    cssTouchAction: 'touchAction',
    cssTransform: 'transform',
    cssTransformBox: 'transformBox',
    cssTransformOrigin: 'transformOrigin',
    cssTransformStyle: 'transformStyle',
    cssTransition: 'transition',
    cssTransitionDelay: 'transitionDelay',
    cssTransitionDuration: 'transitionDuration',
    cssTransitionProperty: 'transitionProperty',
    cssTransitionTimingFunction: 'transitionTimingFunction',
    cssTranslate: 'translate',
    cssUnicodeBidi: 'unicodeBidi',
    cssUnicodeRange: 'unicodeRange',
    cssUserSelect: 'userSelect',
    cssVectorEffect: 'vectorEffect',
    cssVerticalAlign: 'verticalAlign',
    cssViewTransitionName: 'viewTransitionName',
    cssVisibility: 'visibility',
    cssWebkitAlignContent: 'webkitAlignContent',
    cssWebkitAlignItems: 'webkitAlignItems',
    cssWebkitAlignSelf: 'webkitAlignSelf',
    cssWebkitAnimation: 'webkitAnimation',
    cssWebkitAnimationDelay: 'webkitAnimationDelay',
    cssWebkitAnimationDirection: 'webkitAnimationDirection',
    cssWebkitAnimationDuration: 'webkitAnimationDuration',
    cssWebkitAnimationFillMode: 'webkitAnimationFillMode',
    cssWebkitAnimationIterationCount: 'webkitAnimationIterationCount',
    cssWebkitAnimationName: 'webkitAnimationName',
    cssWebkitAnimationPlayState: 'webkitAnimationPlayState',
    cssWebkitAnimationTimingFunction: 'webkitAnimationTimingFunction',
    cssWebkitAppRegion: 'webkitAppRegion',
    cssWebkitAppearance: 'webkitAppearance',
    cssWebkitBackfaceVisibility: 'webkitBackfaceVisibility',
    cssWebkitBackgroundClip: 'webkitBackgroundClip',
    cssWebkitBackgroundOrigin: 'webkitBackgroundOrigin',
    cssWebkitBackgroundSize: 'webkitBackgroundSize',
    cssWebkitBorderAfter: 'webkitBorderAfter',
    cssWebkitBorderAfterColor: 'webkitBorderAfterColor',
    cssWebkitBorderAfterStyle: 'webkitBorderAfterStyle',
    cssWebkitBorderAfterWidth: 'webkitBorderAfterWidth',
    cssWebkitBorderBefore: 'webkitBorderBefore',
    cssWebkitBorderBeforeColor: 'webkitBorderBeforeColor',
    cssWebkitBorderBeforeStyle: 'webkitBorderBeforeStyle',
    cssWebkitBorderBeforeWidth: 'webkitBorderBeforeWidth',
    cssWebkitBorderBottomLeftRadius: 'webkitBorderBottomLeftRadius',
    cssWebkitBorderBottomRightRadius: 'webkitBorderBottomRightRadius',
    cssWebkitBorderEnd: 'webkitBorderEnd',
    cssWebkitBorderEndColor: 'webkitBorderEndColor',
    cssWebkitBorderEndStyle: 'webkitBorderEndStyle',
    cssWebkitBorderEndWidth: 'webkitBorderEndWidth',
    cssWebkitBorderHorizontalSpacing: 'webkitBorderHorizontalSpacing',
    cssWebkitBorderImage: 'webkitBorderImage',
    cssWebkitBorderRadius: 'webkitBorderRadius',
    cssWebkitBorderStart: 'webkitBorderStart',
    cssWebkitBorderStartColor: 'webkitBorderStartColor',
    cssWebkitBorderStartStyle: 'webkitBorderStartStyle',
    cssWebkitBorderStartWidth: 'webkitBorderStartWidth',
    cssWebkitBorderTopLeftRadius: 'webkitBorderTopLeftRadius',
    cssWebkitBorderTopRightRadius: 'webkitBorderTopRightRadius',
    cssWebkitBorderVerticalSpacing: 'webkitBorderVerticalSpacing',
    cssWebkitBoxAlign: 'webkitBoxAlign',
    cssWebkitBoxDecorationBreak: 'webkitBoxDecorationBreak',
    cssWebkitBoxDirection: 'webkitBoxDirection',
    cssWebkitBoxFlex: 'webkitBoxFlex',
    cssWebkitBoxOrdinalGroup: 'webkitBoxOrdinalGroup',
    cssWebkitBoxOrient: 'webkitBoxOrient',
    cssWebkitBoxPack: 'webkitBoxPack',
    cssWebkitBoxReflect: 'webkitBoxReflect',
    cssWebkitBoxShadow: 'webkitBoxShadow',
    cssWebkitBoxSizing: 'webkitBoxSizing',
    cssWebkitClipPath: 'webkitClipPath',
    cssWebkitColumnBreakAfter: 'webkitColumnBreakAfter',
    cssWebkitColumnBreakBefore: 'webkitColumnBreakBefore',
    cssWebkitColumnBreakInside: 'webkitColumnBreakInside',
    cssWebkitColumnCount: 'webkitColumnCount',
    cssWebkitColumnGap: 'webkitColumnGap',
    cssWebkitColumnRule: 'webkitColumnRule',
    cssWebkitColumnRuleColor: 'webkitColumnRuleColor',
    cssWebkitColumnRuleStyle: 'webkitColumnRuleStyle',
    cssWebkitColumnRuleWidth: 'webkitColumnRuleWidth',
    cssWebkitColumnSpan: 'webkitColumnSpan',
    cssWebkitColumnWidth: 'webkitColumnWidth',
    cssWebkitColumns: 'webkitColumns',
    cssWebkitFilter: 'webkitFilter',
    cssWebkitFlex: 'webkitFlex',
    cssWebkitFlexBasis: 'webkitFlexBasis',
    cssWebkitFlexDirection: 'webkitFlexDirection',
    cssWebkitFlexFlow: 'webkitFlexFlow',
    cssWebkitFlexGrow: 'webkitFlexGrow',
    cssWebkitFlexShrink: 'webkitFlexShrink',
    cssWebkitFlexWrap: 'webkitFlexWrap',
    cssWebkitFontFeatureSettings: 'webkitFontFeatureSettings',
    cssWebkitFontSmoothing: 'webkitFontSmoothing',
    cssWebkitHighlight: 'webkitHighlight',
    cssWebkitHyphenateCharacter: 'webkitHyphenateCharacter',
    cssWebkitJustifyContent: 'webkitJustifyContent',
    cssWebkitLineBreak: 'webkitLineBreak',
    cssWebkitLineClamp: 'webkitLineClamp',
    cssWebkitLocale: 'webkitLocale',
    cssWebkitLogicalHeight: 'webkitLogicalHeight',
    cssWebkitLogicalWidth: 'webkitLogicalWidth',
    cssWebkitMarginAfter: 'webkitMarginAfter',
    cssWebkitMarginBefore: 'webkitMarginBefore',
    cssWebkitMarginEnd: 'webkitMarginEnd',
    cssWebkitMarginStart: 'webkitMarginStart',
    cssWebkitMask: 'webkitMask',
    cssWebkitMaskBoxImage: 'webkitMaskBoxImage',
    cssWebkitMaskBoxImageOutset: 'webkitMaskBoxImageOutset',
    cssWebkitMaskBoxImageRepeat: 'webkitMaskBoxImageRepeat',
    cssWebkitMaskBoxImageSlice: 'webkitMaskBoxImageSlice',
    cssWebkitMaskBoxImageSource: 'webkitMaskBoxImageSource',
    cssWebkitMaskBoxImageWidth: 'webkitMaskBoxImageWidth',
    cssWebkitMaskClip: 'webkitMaskClip',
    cssWebkitMaskComposite: 'webkitMaskComposite',
    cssWebkitMaskImage: 'webkitMaskImage',
    cssWebkitMaskOrigin: 'webkitMaskOrigin',
    cssWebkitMaskPosition: 'webkitMaskPosition',
    cssWebkitMaskPositionX: 'webkitMaskPositionX',
    cssWebkitMaskPositionY: 'webkitMaskPositionY',
    cssWebkitMaskRepeat: 'webkitMaskRepeat',
    cssWebkitMaskRepeatX: 'webkitMaskRepeatX',
    cssWebkitMaskRepeatY: 'webkitMaskRepeatY',
    cssWebkitMaskSize: 'webkitMaskSize',
    cssWebkitMaxLogicalHeight: 'webkitMaxLogicalHeight',
    cssWebkitMaxLogicalWidth: 'webkitMaxLogicalWidth',
    cssWebkitMinLogicalHeight: 'webkitMinLogicalHeight',
    cssWebkitMinLogicalWidth: 'webkitMinLogicalWidth',
    cssWebkitOpacity: 'webkitOpacity',
    cssWebkitOrder: 'webkitOrder',
    cssWebkitPaddingAfter: 'webkitPaddingAfter',
    cssWebkitPaddingBefore: 'webkitPaddingBefore',
    cssWebkitPaddingEnd: 'webkitPaddingEnd',
    cssWebkitPaddingStart: 'webkitPaddingStart',
    cssWebkitPerspective: 'webkitPerspective',
    cssWebkitPerspectiveOrigin: 'webkitPerspectiveOrigin',
    cssWebkitPerspectiveOriginX: 'webkitPerspectiveOriginX',
    cssWebkitPerspectiveOriginY: 'webkitPerspectiveOriginY',
    cssWebkitPrintColorAdjust: 'webkitPrintColorAdjust',
    cssWebkitRtlOrdering: 'webkitRtlOrdering',
    cssWebkitRubyPosition: 'webkitRubyPosition',
    cssWebkitShapeImageThreshold: 'webkitShapeImageThreshold',
    cssWebkitShapeMargin: 'webkitShapeMargin',
    cssWebkitShapeOutside: 'webkitShapeOutside',
    cssWebkitTapHighlightColor: 'webkitTapHighlightColor',
    cssWebkitTextCombine: 'webkitTextCombine',
    cssWebkitTextDecorationsInEffect: 'webkitTextDecorationsInEffect',
    cssWebkitTextEmphasis: 'webkitTextEmphasis',
    cssWebkitTextEmphasisColor: 'webkitTextEmphasisColor',
    cssWebkitTextEmphasisPosition: 'webkitTextEmphasisPosition',
    cssWebkitTextEmphasisStyle: 'webkitTextEmphasisStyle',
    cssWebkitTextFillColor: 'webkitTextFillColor',
    cssWebkitTextOrientation: 'webkitTextOrientation',
    cssWebkitTextSecurity: 'webkitTextSecurity',
    cssWebkitTextSizeAdjust: 'webkitTextSizeAdjust',
    cssWebkitTextStroke: 'webkitTextStroke',
    cssWebkitTextStrokeColor: 'webkitTextStrokeColor',
    cssWebkitTextStrokeWidth: 'webkitTextStrokeWidth',
    cssWebkitTransform: 'webkitTransform',
    cssWebkitTransformOrigin: 'webkitTransformOrigin',
    cssWebkitTransformOriginX: 'webkitTransformOriginX',
    cssWebkitTransformOriginY: 'webkitTransformOriginY',
    cssWebkitTransformOriginZ: 'webkitTransformOriginZ',
    cssWebkitTransformStyle: 'webkitTransformStyle',
    cssWebkitTransition: 'webkitTransition',
    cssWebkitTransitionDelay: 'webkitTransitionDelay',
    cssWebkitTransitionDuration: 'webkitTransitionDuration',
    cssWebkitTransitionProperty: 'webkitTransitionProperty',
    cssWebkitTransitionTimingFunction: 'webkitTransitionTimingFunction',
    cssWebkitUserDrag: 'webkitUserDrag',
    cssWebkitUserModify: 'webkitUserModify',
    cssWebkitUserSelect: 'webkitUserSelect',
    cssWebkitWritingMode: 'webkitWritingMode',
    cssWhiteSpace: 'whiteSpace',
    cssWidows: 'widows',
    cssWidth: 'width',
    cssWillChange: 'willChange',
    cssWordBreak: 'wordBreak',
    cssWordSpacing: 'wordSpacing',
    cssWordWrap: 'wordWrap',
    cssWritingMode: 'writingMode',
    cssX: 'x',
    cssY: 'y',
    cssZIndex: 'zIndex',
    cssZoom: 'zoom'
};
